Revise Databricks Iceberg support documentation (#8052)
fishtown-analytics/docs.getdbt.com: website/docs/docs/mesh/iceberg/databricks-iceberg-support.md

| :---- | :---- | :---- | :---- |
| name | Name of the Catalog on Databricks | Yes | “my_unity_catalog” |
| catalog_type | Type of catalog  | Yes | unity, hive_metastore |
| external_volume | Storage location of your data | Optional | See Databricks [documentation](https://docs.databricks.com/aws/en/volumes/managed-vs-external) |
| table_format | Table Format for your dbt models will be materialized as  | Optional | Defaults to `delta` unless overwritten in Databricks account. 

> | adapter_properties: | Additional Platform-Specific Properties.  | Optional | See below for acceptable values	 |

Author: Major revision here 🙏

> | table_format | Table Format for your dbt models will be materialized as  | Optional | Fixed to `iceberg` for `catalog_type=unity` and `default` for `hive_metastore` 

Reviewer: ```suggestion
| table_format | Format for tables created by dbt models.  | Optional | Automatically set to `iceberg` for `catalog_type=unity`; and `default` for `hive_metastore`. |
```

> | file_format | File Format for your dbt models will be materialized as  | Optional | Defaults to `delta` unless overwritten in Databricks account. 

Reviewer: ```suggestion
| file_format | Format used for dbt model outputs.   | Optional | Defaults to `delta` unless overwritten in Databricks account.  |
```

#### Note

> On Databricks, if a model has `catalog_name=<>` in its model config, the catalog name becomes the catalog part of the model's FQN. For example, if the catalog is named my_database, a model with `catalog_name='my_database'` is materialized as `my_database`.<schema>.<model>`.

Author: a nice to have for users

      - name: unity_catalog_integration
        table_format: iceberg
        catalog_type: unity

>         file_format: delta   

Author: parquet can't be used with iceberg tables
